src/psyclone/parse/kernel.py

```diff
@@ -624,29 +633,39 @@ def get_kernel_metadata(name, ast):
         raise ParseError("Kernel type {0} does not exist".format(name))
     return ktype
 
+
 def get_kernel_interface(ast):
     '''Takes the kernel module parse tree and returns the interface part
     of the parse tree.
 
     :param ast: parse tree of the kernel module code
     :type ast: :py:class:`fparser.one.block_statements.BeginSource`
 
-    :returns: Name 'name' of the parse tree of the interface 
-    :rtype: :str `fparser.one.block_statements.Interface.name`
+    :returns: Name of the parse tree of the interface block and
+              the names of the module procedures. Or None, None if
+              there is no interface or the interface has no nodule procedures.
+    :rtype: : `fparser.one.block_statements.Interface.name`,
+              `fparser.one.block_statements.Interface.a.module_procedures` '''
 
-    :returns 'None' if there is no interface, the interface has no name or has no module procedures
-    '''
     iname = None
-    for statement, _ in fpapi.walk(ast, -1):
+    sname = None
+    count = 0
+    for statement, _ in fpapi.walk(ast):
         if isinstance(statement, fparser1.block_statements.Interface):
-#           Check the interfaces assigns module procedures.
+            # count the interfaces, then can be only one!
+            count = count + 1
+            if count >= 2:
+                raise ParseError("Kernel has more than one interface.")
+            # Check the interfaces assigns module procedures.
             if statement.a.module_procedures:
-                iname = statement.name
-#      If implicit interface (no name) set to none as there is no procedure name for PSyclone to use.
-            if iname=='':
-                iname=None
-            break
-    return iname
+                iname = statement.name.lower()
+                sname = statement.a.module_procedures
+                # If implicit interface (no name) set to none as there is no
+                # procedure name for PSyclone to use.
+            if iname == '':
+                iname = None
+    return iname, sname
```

Comment:
Please could you convert each element of `sname` to lowercase - that will reduce the need to keep calling `lower()` later on.

Reply:
OK, done.